module2/Homework2-start/ColorPicker/ContentView.swift

```diff
@@ -0,0 +1,167 @@
+import SwiftUI
+
+struct ContentView: View {
+  @Environment(\.verticalSizeClass) var verticalSizeClass
+  @Environment(\.horizontalSizeClass) var horizontalSizeClass
+
+  @State private var alertIsVisible: Bool = false
+  @State private var redColor: Double = Constants.defaultRedColor
+  @State private var greenColor: Double = Constants.defaultGreenColor
+  @State private var blueColor: Double = Constants.defaultBlueColor
+
+  // When the app first runs, the default for previewColor shows up as white and not these values. Why?
+  @State private var previewColor: Color =  Color(red: Constants.defaultRedColor, green: Constants.defaultGreenColor, blue: Constants.defaultBlueColor)
```

Comment:
You don't divide the value by `255` or your `Constants.colorRange` value. So...a `Color` wants values between 0...1 for each of the color channels and you're giving it numbers that are WAY over 1 so it rounds down to 1. That gives you white.

Reply:
Aha! That explains it and also explains why the same code I had in the init() version wasnt working either. Thanks for catching this!